econml/_ortho_learner.py

```diff
@@ -605,6 +605,7 @@ def fit(self, Y, T, X=None, W=None, Z=None, *, sample_weight=None, sample_var=No
                 self.transformer = FunctionTransformer(
                     func=_EncoderWrapper(self._one_hot_encoder).encode,
                     validate=False)
+                self._set_encoded_treatment_names(self._one_hot_encoder)
```

Comment:
I'm not a super fan of this: This is logic that is at the level of the `TreatmentExpansionMixin`, but now we are pushing it to a lower level, which seems wrong. 

Suppose for instance that someone comes and wants to contribute an estimator. They would need to know that if I expand treatments, then I need to call this function too, but this is not enforced by any library contract.

I think we should think harder if there is a way to push this to the TreatmentExpansionMixin level or to at least make this part of a contract that is enforcible, e.g. that any class that inherits fthe TreatmentExpansionMixin, must also be implementing an `ecnoded_treamtnent_names` and whenever this happens we update the treatment names at the `TreatmentExpansionMixin` level. This way if someone implements an estimator that inherits from this mixin, they'll get an abstract class error, becuase they haven't implemented this.

Reply:
I refactored this and removed much of the old logic due to updates in since sklearn 0.20.